### PR TITLE
plugins.json: fix test for options & defaults on `plugin-response`

### PR DESCRIPTION
**Pull request by achselschweisz**
test: add application/json headers to affected tests, add a mixed default / call test

### CODE REVIEW

**Comment by achselschweisz** [comment]
@skenqbx  please review

File: test/test-http-json.js [hunk 1]
```diff
@@ -32,8 +32,41 @@ suite('http:json', function() {
...
+      assert.strictEqual(response.statusCode, 200);
+
+      assert(!response.buffer);
+      assert(!response.json);
+
+      // we get the response
+      var data_ = [];
+
+      response.on('readable', function() {
+        data_.push(new Buffer(response.read()));
```
Comment (skenqbx) [inline test/test-http-json.js]: `response.read()` should already return a buffer

File: lib/plugins/json.js [hunk 2]
```diff
@@ -21,11 +21,16 @@ module.exports = JSONPlugin;
 JSONPlugin.prototype._setup = function() {
   var self = this;
   var rail = this._rail;
+  var jsonHeader_ = false;
```
Comment (skenqbx) [inline lib/plugins/json.js]: Is always true after the first response with `application/json`

**Comment by skenqbx** [comment]
A tiny nit & a bug, otherwise LGTM

**Pull request merged**